### PR TITLE
Added new Wallet API which returns UTXO statistics

### PR DESCRIPTION
**Pull request by maciejzaleski**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs [hunk 1]
```diff
@@ -1365,6 +1365,68 @@ public IActionResult SyncFromDate([FromBody] WalletSyncFromDateRequest request)
...
+            {
+                return ModelStateErrors.BuildErrorResponse(this.ModelState);
+            }
+
+            try
+            {
+                IEnumerable<UnspentOutputReference> spendableTransactions = this.walletManager.GetSpendableTransactionsInAccount(new WalletAccountReference(request.WalletName, request.AccountName), request.MinConfirmations);
+                
+                model.TotalUtxoCount = spendableTransactions.Count();
+                model.UniqueTransactionCount = (from s in spendableTransactions
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs]: @maciejzaleski we don't really do LINQ queries like this anywhere in the solution, so we tend to stick LINQ like so:

```
model.UniqueTransactionCount = spendableTransactions.GroupBy(s=>s.Transaction.Id).Count();
```

File: src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs [hunk 2]
```diff
@@ -1365,6 +1365,68 @@ public IActionResult SyncFromDate([FromBody] WalletSyncFromDateRequest request)
             return this.Ok();
         }
 
+        [Route("wallet-stats")]
+        [HttpGet]
+        public IActionResult WalletStats([FromQuery] WalletStatsRequest request)
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs]: General question, what's the purpose of this PR is it mainly informational or is it needed by exchanges for example or maybe the test tool?

File: src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs (hunk 1 above)
Comment (maciejzaleski) [reply src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs]: I reject this comment. The code works and is readable.

File: src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs (hunk 2 above)
Comment (maciejzaleski) [reply src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs]: This API provides information about the composition of UTXO within the wallet. It is a very useful functionality during testing but could prove helpful while debugging production issues.

**Comment by maciejzaleski** [comment]
> Please rewrite LINQ queries :)

Rejecting suggested change

**Comment by dangershony** [comment]
>Rejecting suggested change

WTF?

`When editing files, keep new code and changes consistent with the style in the files.`  
https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/Documentation/coding-style.md

As a general rule we do not write such linq style queries!!!

**Pull request merged**